src/lib/storage/disk_size.rb

```diff
@@ -130,6 +130,7 @@ def unit_multiplier(unit)
       #
 
       def +(other)
+        return DiskSize.unlimited if any_operand_unlimited?(other)
```

Comment:
What about using the builtin Ruby support for infinity? Could it help and make some operations simpler?

```Ruby
> irb
irb(main):001:0> Float::INFINITY + 5
=> Infinity
irb(main):002:0> Float::INFINITY - 10
=> Infinity
irb(main):003:0> Float::INFINITY / 10
=> Infinity
```